### DISCUSSION TITLE
ReferenceLine Override Height

### DISCUSSION THREAD

**Discussion by alexevladgabriel** [recharts/recharts#4104] [Q&A] [answered]
I want to set custom y1, y2 to a reference line, but I can't see how to do it.

**Comment by ckifer** [comment]
@alexevladgabriel Reference line is meant to be a straight line, either via `x` or via `y`. You can use the `Curve` component if you'd like to make a line with a specific x1, x2, y1, y2. What do you mean by height?

**Reply by ckifer** [reply]
Actually sorry, with `points` rather than the x/y props

[![Edit simple-line-chart (forked)](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/simple-line-chart-forked-lg48pd?file=%2Fsrc%2FApp.tsx%3A45%2C5)

**Comment by alexevladgabriel** [comment]
![image](https://github.com/recharts/recharts/assets/59282365/7d4e567e-940e-4e76-9587-8240178e457a)

@ckifer I want the reference line to exit the container so I can add a custom element and label on it.

**Reply by ckifer** [reply, marked as answer]
@alexevladgabriel thanks.

Try this 
[![Edit simple-area-chart-reference-line-overflow](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/simple-area-chart-forked-plkchr)

You can use `segment` to override `x` or `y` with `ifOverflow="visible"`
